haystack/retriever/dense.py

```diff
@@ -90,14 +91,12 @@ def __init__(self,
         self.sequence_length = saved_state.encoder_params["sequence_length"]
 
         # Init & Load Encoders
-        self.query_encoder = HFBertEncoder.init_encoder(self.pretrained_model_cfg,
```

Comment:
For init, why dont we just use the Huggingface code for loading the question and context encoders like:

```
from transformers import DPRContextEncoder, DPRContextEncoderTokenizer
tokenizer = DPRContextEncoderTokenizer.from_pretrained('facebook/dpr-ctx_encoder-single-nq-base')
model = DPRContextEncoder.from_pretrained('facebook/dpr-ctx_encoder-single-nq-base', return_dict=True)

and of course similar for DPRQuestionEncoder

Reply:
I have to reiterate on this topic here.

Do we need the DPRConfig when we want to load models from the cloud? To me using the config only makes sense when we want to initiallize a DPR encoder fro training (e.g. with different projection_dim). Is that correct?

@kolk you also said you need to set projection_dim to 0, otherwise the model isnt loaded correctly. Why is projection dim set here to a potentially non zero value? 
So my proposed changes are as follows:
1. remove DPRConfig and projection_dim from the init(). Also remove the self.query_embedding_model and self.passage_embedding_model since it is not used aftewards
2. when we do training we should have projection dim as additional parameter and possibly instantiate a config object

Reply:
yes